src/openrct2/object/ObjectManager.cpp

```diff
@@ -209,7 +215,7 @@ class ObjectManager final : public IObjectManager
         {
             const ObjectRepositoryItem* item = &_objectRepository.GetObjects()[i];
             if (item->LoadedObject != nullptr && IsObjectCustom(item) && item->LoadedObject->GetLegacyData() != nullptr
-                && !item->LoadedObject->IsJsonObject())
+                && item->LoadedObject->GetGeneration() == ObjectGeneration::DAT)
```

Comment:
When the NSF is there, this bit should be removed.

Reply:
Sounds like a TODO comment will be useful here, to make them easier to find.